### PR TITLE
fix: use correct class name and id selector in query()

### LINKED ISSUE #186: [Bug]: `query()` does not return elements correctly
### Checklist before reporting

- [x] I have searched for [similar issues](https://github.com/thalissonvs/pydoll/issues) and didn't find a duplicate.
- [x] I have updated to the latest version of pydoll to verify the issue still exists.

### pydoll Version

2.3.0

### Python Version

3.13.5

### Operating System

Windows

### Bug Description

Using `.query(".classname")` returns no elements.

### Steps to Reproduce

Code example below:

### Code Example

```python
import asyncio

from pydoll.browser import Chrome


async def main():
    async with Chrome() as browser:
        tab = await browser.start()

        await tab.go_to("https://www.wikipedia.org/")

        await asyncio.sleep(2)

        footer = await tab.query(".footer", find_all=True, raise_exc=False)
        print(footer) # does not work - empty list

        footer = await tab.find(class_name="footer", find_all=True, raise_exc=False)
        print(footer) # works


asyncio.run(main())
```

### Additional Context

I've fixed this on my personal fork https://github.com/radiantly/pydoll/tree/fix-query

### PR DESCRIPTION
**Pull request by radiantly**
# Bug Fix Pull Request

## Related Issue(s)
Fixes #186 

## Bug Description & Root Cause
When querying a class `.red`, `_get_expression_type()` returns `By.CLASS_NAME`. Then in `_get_find_elements_command()` it prepends a period resulting in `..red`. Thus, no results are returned.

## Solution
There are two ways to fix this bug.
1. Strip the preceding period and then use `By.CLASS_NAME`
2. Use `By.CSS_SELECTOR`

I opted to fix this using the latter option. This is because `query()` is analogous to `document.querySelector()`, and the selector could look like `.red[data-hover="green"]`. The current logic would try to use `By.CLASS_NAME` because of the preceding period which isn't really correct.

## Verification Steps
Fix works for the code example given in the original issue.

## Testing
The `_get_find_elements_command` isn't great for unit testing because it's doing two things. It's first doing the job of creating the selector (adding the preceding `.` or `#` and then escaping it) and then creating the command. I haven't added unit tests.

## Testing Checklist
- [ ] Added regression test that would have caught this bug
- [ ] Modified existing tests to account for this fix
- [x] All tests pass
- [ ] Edge cases have been tested

## Impact

- [x] Low (isolated fix with no side effects)
- [ ] Medium (might affect closely related functionality)
- [ ] High (affects multiple areas or changes core behavior)

## Backwards Compatibility
- [x] This change is fully backward compatible
- [ ] This change introduces backward incompatibilities (explain below)

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have added test cases that prove my fix is effective
- [x] I have run `poetry run task lint` and fixed any issues
- [x] I have run `poetry run task test` and all tests pass
- [x] My commits follow the [conventional commits](https://www.conventionalcommits.org/) style with message explaining the fix

### CODE REVIEW

File: pydoll/elements/mixins/find_elements_mixin.py [hunk 1]
```diff
@@ -352,18 +351,11 @@ def _get_expression_type(expression: str) -> By:
...
         Patterns:
-        - XPath: starts with //, .// , ./, or /
-        - ID: starts with #
-        - Class: starts with . (but not ./)
+        - XPath: starts with ./, or /
         - Default: CSS_SELECTOR
         """
-        xpath_pattern = r'^(//|\.//|\.\/|/)'
-        if re.match(xpath_pattern, expression):
+        if expression.startswith('./') or expression.startswith('/'):
```
[the remark this answers is not in the document]
Comment (radiantly) [reply pydoll/elements/mixins/find_elements_mixin.py:357]: If it's an xpath that starts with a double slash, it also starts with a single slash.

File: pydoll/elements/mixins/find_elements_mixin.py [hunk 2]
```diff
@@ -435,7 +427,7 @@ def _get_find_elements_command(self, by: By, value: str, object_id: str = ''):
             case _:
                 selector = escaped_value
         if object_id and not by == By.XPATH:
-            script = Scripts.RELATIVE_QUERY_SELECTOR_ALL.replace('{selector}', escaped_value)
```
[the remark this answers is not in the document]
Comment (radiantly) [reply pydoll/elements/mixins/find_elements_mixin.py:438]: We set `selector` to the symbol + escaped_value in the match statement above. I made this change so that it would be consisted with the line 448 below

File: pydoll/elements/mixins/find_elements_mixin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thalissonvs) [reply pydoll/elements/mixins/find_elements_mixin.py:357]: You're right

**Review by thalissonvs — APPROVED** [review]
Thanks :)
